parse/parse.go

```diff
@@ -242,26 +242,119 @@ func (p *parser) parsePatterns() ([]Pattern, error) {
 		switch itm.typ {
 		case itemError:
 			return nil, fmt.Errorf("got error token: '%s'", itm.val)
+		case itemWhitespace:
+			continue
 		case itemText:
 			pattern = append(pattern, TextPattern(itm.val))
 		case itemExpressionOpen:
-			p.next() // skip opening brace
+			// HACK: Find if it's a markup or expression.
+			isMarkup := func() bool {
+				i := 1
+
+				for next := p.next(); ; next = p.next() {
```

Comment:
Do you have to peek into the expression boundaries? E.g. this seems to go outside expression, you may find a closing markup in another expression.

Reply:
Does simple `peekNonWS()` solve it?

```go
    if r := p.peekNonWS(); r ==  itemMarkupOpen || r == itemMarkupOpen {
       continue
    }
```
Markup follows immediately after the expression open.

```abnf
markup       = "{" [s] markup-open *(s attribute) [s] ["/"] "}"
             / "{" [s] markup-close *(s attribute) [s] "}"
markup-open  = "#" identifier *(s option)
markup-close = "/" identifier
```

Reply:
need to test, but i tested many alternatives, this was the one that worked

Reply:
would be nice if lexer gave itemMarkupStart\End, but it also needs some tweaks to make it work